portation of product_brand to 8.0

dulhaver opened the pull request:
coding was done by Seraphine Lantiblé. I am just trying to organize the PR into the OCA:8.0

dulhaver committed:
Update __openerp__.py

set 'installable' to 'True'

dulhaver committed:
Update __init__.py

dulhaver committed:
Update __openerp__.py

dulhaver committed:
Update product_brand.py

dulhaver committed:
Update product_brand_view.xml

dulhaver committed:
Update __init__.py

dulhaver committed:
Update __openerp__.py

dulhaver committed:
Update product_brand.py

dulhaver committed:
Update product_brand_view.xml

dulhaver committed:
ported product_brand

rvalyi commented on product_brand/product_brand_view.xml:
```diff
@@ -71,25 +71,13 @@
-                <field name="categ_id" position="after"  >
-                  <field name="product_brand_id" />
+                <field name="name" position="after"  >
+                  <field name="product_brand_id" placeholder="Marque"/>
```
placeholder shouldn't be in French, it could be "Brand" + a French translation in a po file eventually.

dulhaver replied:
replaced "Marque" with "Brand"

yvaucher commented:
@vrms Thanks for the proposal

There are few things to fix.

You can repair those commits by authoring them correctly

You can do this by doing:
git commit --amend --author="Name email@address.com"

By the way those commit could be squashed plus it would be better to put commit message with some meaning.

"Portation of product_brand to 8.0" would be fine as a commit message

And finally some work is needed to comply with PEP8, see:
https://travis-ci.org/OCA/product-attribute/jobs/46077536

pedrobaeza commented on product_brand/__init__.py:
```diff
@@ -0,0 +1,30 @@
+#####################################################################################
+# Product Brand is an Openobject module wich enable Brand management for products   #
+#####################################################################################
+import product_brand
```
`from . import brand`

dulhaver replied:
changed line 30 "import product_brand"   to   "from . import brand"

dulhaver replied:
changed lenghts of lines in the comments at the begiingin of files as per https://travis-ci.org/OCA/product-attribute/jobs/46077536 for
**init**.py
**openerp**.py
product_brand.py

don't know how to change lengths of lines in the code of product_brand.py because I can not code

don't understand the "missing whitespace after ',' " errors for line 38 & 46. Does not look to me that there are whitespaces missing

pedrobaeza commented on product_brand/__openerp__.py:
```diff
@@ -0,0 +1,57 @@
+    'category': 'Product',
+    'summary' : ' brand',
+    'description': """
+Product Brand Manager
```
Please extract this description (and remove it here) to a file README.rst, because v8 allows it.

pedrobaeza commented:
You can (and should) credit the author of the code via `git commit --author`

pedrobaeza commented:
Please fix the length of the lines for PEP8 compliant.

dulhaver commented:
thanks for the feedback, so far. Will try to fix these issues.I have 0 to nothing experience with all this, so I guess it'll take some time.  Is there any time limit for applying the fixes or can my PR sit a while?

pedrobaeza commented:
Don't worry. There's no time constraint.

dulhaver committed:
line 80, changed "Marque" to "Brand

line 80, changed "Marque" to "Brand

dulhaver committed:
changed line 30

changed "import product_brand" to "from . import brand"

dulhaver committed:
Update __openerp__.py

line 43 removed whitespace in 'category' : 'Product',

reduced length of line to 79

dulhaver committed:
Update __init__.py

line 1 - 30: reduced length of line to 79

dulhaver committed:
Update __init__.py

line 12: removed to whitespace at end of line

dulhaver committed:
Update product_brand.py

line 1-30: reduced length to 79

willArrive committed:
porting product_brand to 8.0, fixed several whitespace issues, added Author

line 1-30: reduced length to 79

willArrive committed:
port product_brand to 8.0, fixes several whitespace errors, added author Seraphine Lantible <s.lantible@gmail.com>

willArrive committed:
port product_brand to 8.0, fixes trailing whitespace and 2 line wrapping errors, added author Seraphine Lantible <s.lantible@gmail.com>

dulhaver commented:
pushed a new commit e91201a just now:
- fixed several whitespace errors as per https://travis-ci.org/OCA/product-attribute/jobs/46077536
- added author --amend --author="author"

I understand "This branch has failed checks, but can be merged" that I can now just lean back & wait for someone with the authority to merge the PR, right?

EDIT: sorry I didn't see the commits from willArrive (who is the actual Author by the way). It seems we did the same work twice now. I hope it's just waist of my time and nothing worse

dulhaver commented:
can anyone let me know whether I still have to do anything here? I was under the impression that "...has failed checks, but can be merged" meant that this was about to be merged.

yvaucher commented:
Still those issues to fix:

```
./product_brand/__openerp__.py:12:80: W291 trailing whitespace
./product_brand/product_brand.py:39:80: E501 line too long (139 > 79 characters)
./product_brand/product_brand.py:47:80: E501 line too long (110 > 79 characters)
```

About authoring, you should do a `git rebase -i origin/8.0` and fix author / squash / edit some commits

willArrive committed:
portation to 8.0. fixes a trailing whitespace and 2 line wrapping errors

willArrive committed:
Merge branch 'prisnet_141221' of https://github.com/vrms/product-attribute into prisnet_141221

dulhaver committed:
Merge branch 'prisnet_141221' of https://github.com/vrms/product-attribute into prisnet_141221

willArrive committed:
Merge branch 'prisnet_141221' of https://github.com/vrms/product-attribute into prisnet_141221

willArrive committed:
Merge branch 'prisnet_141221' of https://github.com/vrms/product-attribute into prisnet_141221

dulhaver commented:
seems I am stuck with fixing the word wraps necessarry to reduce length of lines to 79 in product_brand.py

Funny thing ... I am doing --amend ---author ... now all the time and my (probably silly) mistakes are beeing accounted for someone else ... I am sure she'll like that

dulhaver committed:
removed the line breaks in "product_brand.py" to check whether these where the reason for Travis check to have failed

dulhaver committed:
another attempt to fix length of "prodcut_brand.py" lines to 79 characters

dulhaver committed:
fix word wrapping (in order to meet the 79 characters per line convention)

dulhaver committed:
fixing word wraps

dulhaver commented:
I am a little at the end of the line. Seems I can't manage to meet requirements and fix the word wraps in a way travis checks do not fail. Any ideas how to proceed? Do I withdraw my PR and use the failed initial code (which works) in a 'private' Module?

dulhaver committed:
removes trailing whitespaces

the trailing whitespaces originated from the previous shortening of line lenght to meet 79 character rule

yvaucher commented:
@vrms ok it seems you need some help here

To write this on a 80 char basis:

```
'product_brand_id': fields.many2one('product.brand', 'Brand', help='Select a brand for this product.')
```

You have to:
- put it on multiple lines
- as it is between parenthesis you don't need line break (`\`)
- you have to indent it, you have 2 choices:
  - at 4 spaces if first param is on second line

.

```
    'product_brand_id': fields.many2one(
        'product.brand', 'Brand',
        help='Select a brand for this product.')
```
- or at level of opening parenthesis

.

```
    'product_brand_id': fields.many2one('product.brand', 'Brand',
                                        help='Select a brand for this product.')
```

dulhaver committed:
Update product_brand.py

dulhaver committed:
Update product_brand.py

dulhaver committed:
Update product_brand.py

dulhaver committed:
added a comma at the end of line 43 in product_brand.py

dulhaver commented:
looks like it is not getting better ... I guess I have to go back to school and come back in a while

dulhaver committed:
Update product_brand.py

yvaucher commented on product_brand/__openerp__.py:
```diff
@@ -0,0 +1,58 @@
+# Mathieu Lemercier, mathieu@netandco.net, #                                  #
+# Franck Bret, franck@netandco.net #                                          #
+# Seraphine Lantible, s.lantible@gmail.com, http://www.prisnet.ch             #
+# #                                                                           # 
```
@vrms You are almost done for pep8. You just have an extra space on this line as stated in travis:

```
./product_brand/__openerp__.py:12:80: W291 trailing whitespace
```

dulhaver replied:
thanks, yannick. 

Seems my problem is that I do not read the Travis reports. Actually I don't know where to find the details.  All I ever see is '...has failed checks'. Seems to me Travis gives you much more details but I just don't see where those are shown.

Now, with the removed qhitespace 1 of 3 checks did not fail. The total message still ist '... has failed checks' ... I guess you can make sense to that. Whether you have a minute to explain I'll promise to use this knowledge to help on OCA projects

dulhaver committed:
removed trailing white space in line 12

yvaucher commented:
@vrms You have the detail button on right of "X Failed -- The Travis CI build failed"

Or you can click on the red X next to commit ref to open details.

There are 3 tests https://travis-ci.org/OCA/product-attribute/builds/47864120

1 is test for coding style conventions
1 is for tests with OCB branch
1 is for tests with official Odoo branch

The error is:

```
 File "/home/travis/build/OCA/product-attribute/product_brand/__init__.py", line 31, in <module>

from . import brand

ImportError: cannot import name brand
```

yvaucher commented on product_brand/__init__.py:
```diff
@@ -0,0 +1,31 @@
+# Product Brand is an Openobject module wich enable Brand management          #
+# for products                                                                #
+###############################################################################
+from . import brand
```
Here you import `brand` but file is `product_brand.py`

So you need to write:
`from . import product_brand`

dulhaver replied:
@yvaucher: I know the "Details" button, but I don't get to see the details of the checks when I click there. I see there are 3 builds and I see that they fail (actually the last one I did had one passing and 2 failing).
Can it be that team members of a repo (like you) get to see more infos then non-members (like me)? If I did the Travis from inside of my fork, then I might be able to get more details

dulhaver committed:
Update __init__.py

yvaucher commented:
And we are green!

:+1: 

Ping me once this get a second approval, I'll do the commit --author and squashes to show correctly what you fixed and what @willArrive did.

dulhaver commented:
@yvaucher: I am not sure that you will find any of the ported code on the "willArrive" github profile. Seraphine sent me her 8.0 version by email. We had a skype session where I tried to walk her through the procedure do the PR to OCA/product-attribute from her own repo but that didn't work out due to communication problems. The credit for the code entirely goes to her I'd say. I don't even understand the code really

dulhaver commented:
@pedrobaeza as I understand there needs to be a 2nd review before this can be merged. Would you mind?

pedrobaeza commented on product_brand/__openerp__.py:
```diff
@@ -0,0 +1,58 @@
+    'category': 'Product',
+    'summary': ' brand',
+    'description': """
+Product Brand Manager
```
Move this to a README.rst file for GitHub+Odoo compatibility.

pedrobaeza commented:
A little remark. The rest LGTM.

dulhaver committed:
Create README.rst

moved line 38-58 from __openerp__.py to this README

dulhaver committed:
Update __openerp__.py

removed line 38 - 58 and put them into a new README.rst

pedrobaeza commented:
You have to move entirely the description (including title), and remove the key from the manifest.

dulhaver committed:
Update README.rst

dulhaver committed:
Update __openerp__.py

removed line 31 - 38 (which moved to README.rst as well)

dulhaver committed:
Update __openerp__.py

dulhaver committed:
added back lines 32 - 54

removing them to the README made the travis fail

dulhaver committed:
added indentation

dulhaver committed:
Update __openerp__.py

dulhaver committed:
Update __openerp__.py

dulhaver commented:
@pedrobaeza I don't think that is precisely what you suggested to do. But at least the travis doesn't fail now. "...remove the key from the manifest." is what I did not follow because I don't understand. 

'manifest' is the   **openerp**.py    I think 
not clear what    'the key'    could be

pedrobaeza commented:
That is python terminology. Indeed the manifest file is `__openerp__.py`, and the key is the dictionary entry:

```
{
    ...
    'author': ...,
    # Remove this key
    'description': """...."""
    # ############
    'installable': ...,
}
```

dulhaver commented:
@pedrobaeza I don't think that is precisely what you suggested to do. But at least the travis doesn't fail now. "...remove the key from the manifest." is what I did not follow because I don't understand. 

'manifest' is the   **openerp**.py    I think 
not clear what    'the key'    could be

yvaucher commented:
@vrms `__openerp__.py` contains a python dictionary `{...}` composed of `keys` and `values`

`description` is a key. The idea is to remove that key from dict to move it's value 

```
Product Brand Manager
===================
...
```

In a file named `README.rst` like in this template:

https://github.com/OCA/maintainers-tools/blob/master/template/module/README.rst (open it in raw to see the format)

We do this to have the description available and formated for both Odoo and github.

Here is an example of module in which we already did it:
https://github.com/OCA/stock-logistics-workflow/tree/8.0/transport_information
https://github.com/OCA/stock-logistics-workflow/blob/8.0/transport_information/__openerp__.py
https://github.com/OCA/stock-logistics-workflow/blob/8.0/transport_information/README.rst

dulhaver committed:
Update README.rst

dulhaver committed:
removed description (line 36-47)

yvaucher commented:
I cleaned this PR under: https://github.com/OCA/product-attribute/pull/48
